### PR TITLE
1640: Update and bundle SWIG

### PR DESCRIPTION
**Pull request by tst-cjeffries**
Update and bundle a newer version swig to allow full migration of C++11.

To test (on EL6, and Ubuntu):
- Build portable: You should see swig build in the output at this step.
- Install portable
- Verify that it is working.
- Build and install open.
- Verify that it works.

Also review changes to the documentation and the GitHub portable wiki page.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -6,3 +6,4 @@
 *.glc filter=lfs diff=lfs merge=lfs -text
 *.glm filter=lfs diff=lfs merge=lfs -text
 earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/test_data/** binary
+earth_enterprise/third_party/swig/swig-4.0.1.tar.gz filter=lfs diff=lfs merge=lfs -text
```
Comment (tst-rwildes) [inline .gitattributes:9]: The `*.gz` above doesn't cover this?

File: earth_enterprise/src/fusion/gepublish/SConscript [hunk 2]
```diff
@@ -46,7 +47,7 @@ swig_env['CPPFLAGS'] += ['-fPIC']  # Needed for 64-bit compilation.
 swig_env['CPPFLAGS'] += ['-I%s' % swig_env['python_include']]
 swig_env.Append(SWIGFLAGS = ['-python', '-c++'])
 
-libgepublishmanagerhelper = swig_env.sharedLib('gepublishmanagerhelper',
+libgepublishmanagerhelper = swig_env.SharedLibrary('gepublishmanagerhelper',
```
Comment (tst-rwildes) [inline earth_enterprise/src/fusion/gepublish/SConscript:50]: What is this change doing?

File: earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/shared/file_unpacker.h [hunk 3]
```diff
@@ -35,16 +35,6 @@ class GlcReader;
  * Class for unpacking files from a single package file.
  */
 class FileUnpacker {
-  /**
```
Comment (tst-rwildes) [inline earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/shared/file_unpacker.h:38]: You should add these back in but as deleted and public.  Also should do the same for move operations

File: earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/build_and_test.py [hunk 4]
```diff
@@ -91,6 +99,8 @@ def BuildLibrary(os_dir, ignore_results):
     archData = platform.architecture(pathToLib)
     if archData[0] == "64bit":
       specialDefs = "-DMS_WIN64"
+  elif os_dir == "Linux":
+    build_swig(source_dir)
```
Comment (tst-rwildes) [inline earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/build_and_test.py:103]: We only need to do this on Linux?  What about Mac or Windows?

File: .gitattributes (hunk 1 above)
Comment (tst-cjeffries) [reply .gitattributes:9]: I bet the git lfs add added this. Shall I remove then?

File: earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/build_and_test.py (hunk 4 above)
Comment (tst-cjeffries) [reply earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/build_and_test.py:103]: I honestly don't know about mac, but windows we have the newer version of swig available to us, so we don't need to build it.

File: earth_enterprise/src/fusion/gepublish/SConscript (hunk 2 above)
Comment (tst-cjeffries) [reply earth_enterprise/src/fusion/gepublish/SConscript:50]: Ah. Yeah that needs changed back to sharedlib. Sharedlibrary is the scons builder, and sharedLib appears to be our kh version.

**Comment by tst-cjeffries** [comment]
Its also worth noting that the wiki page will need to be updated for portable. I will do this once the PR is merged in, as it isn't a static file that can be edited as part of the PR.

File: earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/shared/file_unpacker.h (hunk 3 above)
Comment (tst-cjeffries) [reply earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/shared/file_unpacker.h:38]: Misunderstood the TODO. Thanks.

Comment (tst-cjeffries) [reply earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/shared/file_unpacker.h:38]: Added back is a deleted and public.

File: earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/build_and_test.py (hunk 4 above)
Comment (tst-rwildes) [reply earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/build_and_test.py:103]: Looks like according to docs Portable is not supported on Mac so we can forget about that.  I do thing the docs need to be updated to reflect these changes: https://www.opengee.org/geedocs/5.3.4/answer/portable/portableUserGuideWinLinux.html

Comment (tst-cjeffries) [reply earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/build_and_test.py:103]: Ok. That seems logical to me. Ill make reviewing the changes to the wiki part of reviewing the PR.

File: earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/build_and_test.py [hunk 5]
```diff
@@ -45,6 +45,14 @@
 import opengee.version
 
 
+def build_swig(source_dir):
```
Comment (tst-rwildes) [inline earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/build_and_test.py:48]: IDK...  At the end of the day should we just say this script is for Windows building of Portable only.  And give a scons target for building portable when in Linux?

Comment (tst-cjeffries) [reply earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/build_and_test.py:48]: I agree, but that would be quite a major refactor. For the time being I believe it makes the most sense to make those changes as a separate issue.

**Pull request merged**